libraries/abstractions/common_io/test/test_iot_pwm.c

```diff
@@ -420,109 +400,42 @@ TEST( TEST_IOT_PWM, AFQP_IotPwmAccuracy )
 
-    lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioPull, &xGpioPullB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
-
-    lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioInterrupt, &xGpioInterruptB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
-
-    iot_gpio_set_callback( xtestIotGpioHandleB, prvPwmGpioInterruptCallback, &xGpioInterruptB );
+        lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioPull, &xGpioPullB );
+        TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
 
-    /* Initialize our counter to 0 */
-    ultestIotPwmIrqCounter = 0;
-
-    /* Open PWM handle */
-    xPwmHandle = iot_pwm_open( ultestIotPwmInstance );
-    TEST_ASSERT_NOT_EQUAL( NULL, xPwmHandle );
+        lRetVal = iot_gpio_ioctl( xtestIotGpioHandleB, eSetGpioInterrupt, &xGpioInterruptB );
+        TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
 
-    /* Settup the pwm configuration */
-    xSetPwmConfig.ulPwmFrequency = ultestIotPwmFrequency;
-    xSetPwmConfig.ucPwmDutyCycle = ultestIotPwmDutyCycle;
-    xSetPwmConfig.ucPwmChannel = ultestIotPwmChannel;
+        iot_gpio_set_callback( xtestIotGpioHandleB, prvPwmGpioInterruptCallback, &xGpioInterruptB );
 
-    /* Set the pwm configuration */
-    lRetVal = iot_pwm_set_config( xPwmHandle, xSetPwmConfig );
-    TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
+        /* Initialize our counter to 0 */
+        ultestIotPwmIrqCounter = 0;
 
-    /* Start the PWM signal */
-    lRetVal = iot_pwm_start( xPwmHandle );
-    TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
+        /* Open PWM handle */
+        xPwmHandle = iot_pwm_open( ultestIotPwmInstance );
+        TEST_ASSERT_NOT_EQUAL( NULL, xPwmHandle );
 
-    /* Delay for 5 sec. */
-    vTaskDelay( PWM_TEST_DURATION_MSEC / portTICK_PERIOD_MS );
+        /* Settup the pwm configuration */
+        xSetPwmConfig.ulPwmFrequency = ultestIotPwmFrequency;
+        xSetPwmConfig.ucPwmDutyCycle = ultestIotPwmDutyCycle;
+        xSetPwmConfig.ucPwmChannel = ultestIotPwmChannel;
 
-    /* Close PWM handle */
-    lRetVal = iot_pwm_close( xPwmHandle );
-    TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
+        /* Set the pwm configuration */
+        lRetVal = iot_pwm_set_config( xPwmHandle, xSetPwmConfig );
+        TEST_ASSERT_EQUAL( IOT_PWM_SUCCESS, lRetVal );
 
-    /* Close the GPIO handle */
-    lRetVal = iot_gpio_close( xtestIotGpioHandleB );
-    TEST_ASSERT_EQUAL( IOT_GPIO_SUCCESS, lRetVal );
-
-    expected_count = ultestIotPwmFrequency * ( PWM_TEST_DURATION_MSEC / 1000 );
-    TEST_ASSERT_GREATER_THAN( ( expected_count - 100 ), ultestIotPwmIrqCounter );
-    TEST_ASSERT_GREATER_THAN( ultestIotPwmIrqCounter, ( expected_count + 100 ) );
-}
-
-/*-----------------------------------------------------------*/
-
-/**
- * @brief Assited Test Function to test pwm accuracy by running the
- * pwm signal for 5 seconds and also counting the number of pulses
- *
- */
-TEST( TEST_IOT_PWM, AFQP_IotPwmAccuracyAssisted )
```

Comment:
Please keep "assisted" test cases. 

Assisted test cases are expected to not be called in TEST_GROUP_RUNNER. They are supposed to be called via other mechanism.

Reply:
Is it not an option to call AFQP_IotPwmAccuracy via the other mechanism right away? The two tests was identical to each other in every way except the name of the global intance/gpio variable.

Reply:
Please keep the test name as there are other scripts that depend on this test name.

Reply:
The assisted test case has now been added back. refactored that and the non-assisted version so that they share the same test code.